plugin/src/main/java/org/opensearch/ml/model/MLModelCache.java

```diff
@@ -76,17 +76,34 @@ public void clear() {
     }
 
     public void addModelInferenceDuration(double duration, long maxRequestCount) {
-        while (modelInferenceDurationQueue.size() >= maxRequestCount) {
-            modelInferenceDurationQueue.poll();
-        }
-        this.modelInferenceDurationQueue.add(duration);
+        addInferenceDuration(duration, maxRequestCount, modelInferenceDurationQueue);
     }
 
     public void addPredictRequestDuration(double duration, long maxRequestCount) {
-        while (predictRequestDurationQueue.size() >= maxRequestCount) {
-            predictRequestDurationQueue.poll();
+        addInferenceDuration(duration, maxRequestCount, predictRequestDurationQueue);
+    }
+
+    private void addInferenceDuration(double duration, long maxRequestCount, Queue<Double> queue) {
+        resizeInferenceQueue(maxRequestCount, queue);
+        if (maxRequestCount > 0) {
+            queue.add(duration);
+        }
+    }
+
+    public void resizeMonitoringQueue(long maxRequestCount) {
+        log.debug("resize inference duration monitoring queue with size {}", maxRequestCount);
+        resizeInferenceQueue(maxRequestCount, predictRequestDurationQueue);
+        resizeInferenceQueue(maxRequestCount, modelInferenceDurationQueue);
+    }
+
+    private void resizeInferenceQueue(long maxRequestCount, Queue<Double> queue) {
```

Comment:
may be the change the name as this is also resizing `predictRequestDurationQueue`?

Reply:
Both `predictRequestDurationQueue` and `modelInferenceDurationQueue` are for inferencing, I think should be ok.